rts/Sim/Units/UnitTypes/Builder.cpp

```diff
@@ -708,8 +708,25 @@ bool CBuilder::StartBuild(BuildInfo& buildInfo, CFeature*& feature, bool& inWait
 	const CGameHelper::BuildSquareStatus tbs = CGameHelper::TestUnitBuildSquare(buildInfo, feature, -1, true);
 
 	switch (tbs) {
-		case CGameHelper::BUILDSQUARE_OPEN:
-			break;
+		case CGameHelper::BUILDSQUARE_OPEN: {
+			// <pos> might map to a non-blocking portion
+			// of the buildee's yardmap, fallback check
+			const CUnit* u = CGameHelper::GetClosestFriendlyUnit(nullptr, buildInfo.pos, 8.0f, allyteam);
+			if (u != nullptr) {
+				if (CanAssistUnit(u, buildInfo.def)) {
+					// StopBuild sets this to false, fix it here if picking up the same buildee again
+					terraforming = (u == prvBuild && u->terraformLeft > 0.0f);
+
+					AddDeathDependence(curBuild = const_cast<CUnit*>(u), DEPENDENCE_BUILD);
+					ScriptStartBuilding(u->pos, false);
+					return true;
+				}
+
+				// let BuggerOff handle this case (TODO: non-landed aircraft should not count)
+				if (buildInfo.FootPrintOverlap(u->pos, u->GetFootPrint(SQUARE_SIZE * 0.5f)))
+					return false;
+			}
+		} break;
```

Comment:
Why is there BUILDSQUARE_OPEN returned in the first place?

Reply:
See the comment: `// <pos> might map to a non-blocking portion of the buildee's yardmap`

Reply:
Though that comment seems incorrect since ZK windgens don't have a non-blocking portion of the yardmap?

https://github.com/ZeroK-RTS/Zero-K/blob/40e6defe2425b46518cd37105c2b6380db4f920b/units/energywind.lua#L46

Reply:
no, this comment was copied as is from below

Reply:
> Why is there BUILDSQUARE_OPEN returned in the first place?

Basically because CGameHelper::TestUnitBuildSquare is wrong in rare but reproducible cases. See the second image in the OP. You can remove this whole change if you want.

Reply:
Have you tried to debug CGameHelper::TestUnitBuildSquare and fix it?